core/src/Kokkos_View.hpp

```diff
@@ -1749,6 +1749,20 @@ class View : public ViewTraits<DataType, Properties...> {
     m_map   = std::move(rhs.m_map);
     return *this;
   }
+  // -----------------------------------------------------------
+  // public is_assignable
+  // expose underlying view mapping property
+  template <class RT, class... RP>
+  inline const bool is_assignable(const View<RT, RP...>& rhs) {
+    using SrcTraits = typename View<RT, RP...>::traits;
+    using Mapping   = Kokkos::Impl::ViewMapping<traits, SrcTraits,
+                                              typename traits::specialize>;
+
+    if (Mapping::is_assignable)
```

Comment:
Shouldn't this check just be inside of `assignable_layout_check`? When would you ever want to perform the logic inside of that method if you already know the result is `false`?